lib/request.coffee

```diff
@@ -97,6 +103,32 @@ module.exports = getRequest = ({ token, debug = false, retries = 0, isBrowser =
 
 			return options
 
+	interceptRequestOptions = (requestOptions) ->
+		interceptRequestOrError(Promise.resolve(requestOptions))
+
+	interceptRequestError = (requestError) ->
+		interceptRequestOrError(Promise.reject(requestError))
+
+	interceptResponse = (response) ->
+		interceptResponseOrError(Promise.resolve(response))
+
+	interceptResponseError = (responseError) ->
+		interceptResponseOrError(Promise.reject(responseError))
+
+	interceptRequestOrError = (initialPromise) ->
+		Promise.resolve(
+			exports.interceptors.reduce (promise, interceptor) ->
+				promise.then(interceptor.request, interceptor.requestError)
+			, initialPromise
+		)
+
+	interceptResponseOrError = (initialPromise) ->
+		Promise.resolve(
+			exports.interceptors.slice().reverse().reduce (promise, interceptor) ->
+				promise.then(interceptor.response, interceptor.responseError)
```

Comment:
OK it works but shows a warning: https://monosnap.com/file/1uCMR5wTIKJtfoxQwm90SmJtKosfTE.png

So would change to

```
{ response, responseError } = interceptor 
return if (response or responseError) then promise.then(interceptor.response, interceptor.responseError) else promise`
```

(and same below)